prowler/providers/aws/services/awslambda/awslambda_service.py

```diff
@@ -63,27 +64,42 @@ def __list_functions__(self, regional_client):
 
     def __get_function_code__(self):
         logger.info("Lambda - Getting Function Code...")
-        try:
-            for function in self.functions.values():
-                regional_client = self.regional_clients[function.region]
-                function_information = regional_client.get_function(
-                    FunctionName=function.name
-                )
-                if "Location" in function_information["Code"]:
-                    code_location_uri = function_information["Code"]["Location"]
-                    raw_code_zip = requests.get(code_location_uri).content
-                    function_code = LambdaCode(
-                        location=code_location_uri,
-                        code_zip=zipfile.ZipFile(io.BytesIO(raw_code_zip)),
+        # Use a thread pool handle the queueing and execution of the __fetch_function_code tasks, up to max_workers tasks concurrently.
+        with ThreadPoolExecutor(max_workers=10) as executor:
```

Comment:
Do you need this if we first merge the threading in the AWS Service? Because you might use the `ThreadPoolExecutor` present in the AWS class right?

Reply:
We will still need this, because in this case Im yielding the results from the threads so that the code that checks for secrets within the function code can get functions 1-by-1, instead of collecting all the lambda code before running the check (which resulted in large amounts of code stored in memory which was an issue for 1 user who had 44GB of lambda code in their account). With the thread pool, we will have at most 10 function's code in memory at a time, and get the performance benefits of threading the calls (which would take a while to retrieve the code for each function)